File: server/Resseract/resseract-server/src/main/java/abistech/resseract/data/summary/DataSummary.java
package abistech.resseract.data.summary;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.ToString;

import java.util.List;

@AllArgsConstructor
@Getter
@ToString
@EqualsAndHashCode
public class DataSummary {
    private final int rowCount;
    private final int columnCount;
    private final List<Object[]> head;
    private List<ColumnStatistics<?>> columnStatistics;
}

File: angular-app/Resseract/src/app/specs/data-summary.ts
export class DataSummary {
    constructor(
        public rowCount: number,
        public columnCount: number,
        public head: any[][],
        public columnStatistics: ColumnStatistics[],
    ) {
    }
}


export class ColumnStatistics {
    constructor(
        public name: string,
        public type: string,
        public uniqueValueCount: number,
        public nullValueCount: number,
        public valueCount: any[],
    ) {
    }
}